src/screens/profile/CreateAmenities/CreateAmenities.tsx

```diff
@@ -0,0 +1,193 @@
+import { RootStackParamList } from 'src/navigation';
+import { useAppDispatch } from 'src/store';
+import { AsyncThunks } from 'src/store/thunks';
+import { BUTTON_SIZES } from 'src/styles';
+import { ApiSuccessResponseType, IconName } from 'src/types';
+import { AccommodationAmenitiesResponse, UpdateAmenitiesParams } from 'src/types/amenities';
+
+import { styles } from './CreateAmenities.styles';
+import { SelectedAmenities, amenitiesObj, defaultAmenitiesState } from './CreateAmenities.utils';
+
+type CreateAmenitiesNavigationProp = StackNavigationProp<RootStackParamList, 'CreateAmenities'>;
+type CreateAmenitiesRouteProp = RouteProp<RootStackParamList, 'CreateAmenities'>;
+
+export interface AmenitiesProps {
+  navigation: CreateAmenitiesNavigationProp;
+  route: CreateAmenitiesRouteProp;
+}
+
+const CreateAmenities = ({ route }: AmenitiesProps) => {
+  const { isNew, accomodationId } = route.params as { isNew: boolean; accomodationId: string };
+  const dispatch = useAppDispatch();
+  const navigation = useNavigation<NavigationProp<RootStackParamList>>();
+
+  const [selectedAmenities, setSelectedAmenities] =
+    useState<SelectedAmenities>(defaultAmenitiesState);
+  const [otherAmenities, setOtherAmenities] = useState<string[]>([]);
+  const [inputValue, setInputValue] = useState('');
+
+  const amenities: string[] = Object.keys(selectedAmenities);
+  const scrollViewRef = useRef<ScrollView>(null);
+
+  const scrollToBottom = () => {
+    if (scrollViewRef.current) {
+      scrollViewRef.current?.scrollToEnd({ animated: true });
+    }
+  };
+
+  const toggleAmenity = (amenityKey: string) => {
+    setSelectedAmenities((prevSelectedAmenities: SelectedAmenities) => ({
+      ...prevSelectedAmenities,
+      [amenityKey]: !prevSelectedAmenities[amenityKey],
+    }));
+  };
+
+  const addOtherAmenities = () => {
+    if (inputValue.trim() !== '' && inputValue.trim().length < 24) {
+      setOtherAmenities((prevAmenity) => [...prevAmenity, inputValue]);
+      scrollToBottom();
+      setInputValue('');
+    }
+  };
+
+  const renderAdditionalChips = () => {
+    return otherAmenities.map((item, index) => (
+      <Chip
+        key={index}
+        iconName={IconName.Check}
+        iconSet="ionicons"
+        text={item}
+        onTouchFunction={() => removeAdditionalChip(index)}
+        state
+      />
+    ));
+  };
+
+  const removeAdditionalChip = (index: number) => {
+    const updatedOptions = [...otherAmenities];
+    updatedOptions.splice(index, 1);
+    setOtherAmenities(updatedOptions);
+  };
+
+  const onFormSubmit = async () => {
+    try {
+      const response = await dispatch(
+        AsyncThunks.addAmenitiesThunk({
+          accomodationId,
+          data: {
+            ...selectedAmenities,
+            otherAmenities: otherAmenities.join(' ,'),
+          },
+        } as UpdateAmenitiesParams)
+      );
+      if (response.meta.requestStatus === 'fulfilled') {
+        console.log('Amenities added successfully');
```

Comment:
Would you like to handle successful result and reflect it in state?

Reply:
this is boilerplate for next steps of adding/updating amenities inside future 'stepper' feature and from accomodation page. It will be corrected as soon as connected features will be done.